From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Stephen <stephen@vamrs.com>
Date: Thu, 2 Jun 2022 14:25:50 +0800
Subject: [PATCH] arm64: dts: rockchip: rock 5a: Fix ethernet cannot get IP

Here change phy-mode to "rgmii" since tx delay and rx delay
are disabled in hardware.

Signed-off-by: Stephen <stephen@vamrs.com>
Signed-off-by: Yuntian Zhang <yt@radxa.com>
---
 arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts b/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts
index fad83b58558e..5c1a1bc53c40 100644
--- a/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts
@@ -590,14 +590,13 @@ rgmii_phy1: phy@1 {
 
 &gmac1 {
 	status = "okay";
-	// Use rgmii-rxid mode to disable rx delay inside Soc
-	phy-mode = "rgmii-rxid";
+	phy-mode = "rgmii";
 	clock_in_out = "output";
 
 	snps,reset-gpio = <&gpio3 RK_PB7 GPIO_ACTIVE_LOW>;
 	snps,reset-active-low;
 	// Reset time is 20ms, 100ms for rtl8211f
-	snps,reset-delays-us = <0 100000 100000>;
+	snps,reset-delays-us = <0 20000 100000>;
 
 	pinctrl-names = "default";
 	pinctrl-0 = <&gmac1_miim
@@ -606,8 +605,8 @@ &gmac1_rx_bus2
 		     &gmac1_rgmii_clk
 		     &gmac1_rgmii_bus>;
 
-	// tx_delay = <0x45>;
-	// rx_delay = <0x3f>;
+	tx_delay = <0x3a>;
+	rx_delay = <0x3e>;
 
 	phy-handle = <&rgmii_phy1>;
 };
-- 
2.36.1

